### PR TITLE
Add docker validation function

### PR DESCRIPTION
**Pull request by YuvalFeldman**

### CODE REVIEW

File: pkg/k3d/registry.go [hunk 1]
```diff
@@ -249,3 +261,74 @@ func CacheImageInTheBackground(ctx context.Context, image string) error {
...
 	return dockerClient.ContainerExecStart(ctx, exec.ID, dockerTypes.ExecStartCheck{})
 }
+
+func CheckDockerRequirements() error {
+	if os.Getenv("DISABLE_DOCKER_CHECKS") == "true" {
+		return nil
+	}
+	_, err := exec.LookPath("docker")
+	if err != nil {
+		return errors.New("Docker is not installed. Docker is required, please install it and retry. https://docs.docker.com/engine/install/")
```
Comment (moshe-kabala) [inline pkg/k3d/registry.go:271]: consider: Docker is required => Docker is prerequisite

**Review by moshe-kabala — APPROVED** [review]
LGTM

File: pkg/k3d/registry.go [hunk 2]
```diff
@@ -249,3 +261,74 @@ func CacheImageInTheBackground(ctx context.Context, image string) error {
...
+		noResources = true
+	}
+
+	if dockerFreeStorageKB < REQUIRED_STORAGE_KB {
+		log.Printf("Please increase docker storage limit, tensorleap required at least %s free storage\n", REQUIRED_STORAGE_PRETTY)
+		noResources = true
+	}
+
+	if noResources {
+		fmt.Println("Please retry installation after updating your docker config.")
```
Comment (moshe-kabala) [inline pkg/k3d/registry.go:329]: use log

File: pkg/k3d/registry.go [hunk 3]
```diff
@@ -249,3 +261,74 @@ func CacheImageInTheBackground(ctx context.Context, image string) error {
...
+	}
+	dockerMemory := int64(0)
+	if err := json.Unmarshal(dockerMemoryBytes, &dockerMemory); err != nil {
+		log.Fatalf("Failed parsing memory data, %s", err)
+		return err
+	}
+	dockerMemoryPretty := fmt.Sprintf("%dGb", dockerMemory/(1024*1024*1024))
+	log.Printf("Docker has %s memory available.\n", dockerMemoryPretty)
+
+	fmt.Println("Checking docker storage limits...")
```
Comment (moshe-kabala) [inline pkg/k3d/registry.go:295]: use log

File: pkg/k3d/registry.go (hunk 1 above)
Comment (YuvalFeldman) [reply pkg/k3d/registry.go:271]: Updated

File: pkg/k3d/registry.go (hunk 2 above)
Comment (YuvalFeldman) [reply pkg/k3d/registry.go:329]: Updated

File: pkg/k3d/registry.go (hunk 3 above)
Comment (YuvalFeldman) [reply pkg/k3d/registry.go:295]: Update

**Pull request merged**